packages/dashboard/src/components/elements/account-wizard/account-wizard.tsx

```diff
@@ -0,0 +1,284 @@
+                                    {item.header.title}
+                                </Text>
+                            </div>
+                        </MobileWrapper>
+                    </div>
+                ))}
+            </div>
+            <div className='dw-account-wizard__header-active-title'>
+                <Text size='m' weight='bold' styles={{ lineHeight: '3.6rem' }}>
+                    {items[current_step]?.header.active_title}
+                </Text>
+            </div>
+        </div>
+    );
+};
+
+type TAccountWizard = {
+    fetchResidenceList?: () => void;
+    fetchStatesList?: () => void;
+    fetchFinancialAssessment?: () => void;
+    needs_financial_assessment?: () => boolean;
+    has_currency?: () => boolean;
+    has_real_account?: () => boolean;
+    account_settings?: GetSettings;
+    [x: string]: any;
+};
+
+const AccountWizard: React.FC<TAccountWizard> = observer((props: TAccountWizard) => {
+    const [form_error, setFormError] = React.useState('');
+    const [previous_data, setPreviousData] = React.useState([]);
+    const [state_items, setStateItems] = React.useState<TWizardItemConfig[]>([]);
+    const [has_previous_data, setHasPreviousData] = React.useState(false);
+    const [mounted, setMounted] = React.useState(false);
+    const isMounted = useIsMounted();
+    const {
+        fetchStatesList,
+        fetchResidenceList,
+        fetchFinancialAssessment,
+        account_settings,
+        has_currency,
+        has_real_account,
+        is_loading,
+    } = props;
+
+    React.useEffect(() => {
+        if (!account_settings) return;
+
+        Promise.all([fetchStatesList?.(), fetchResidenceList?.(), fetchFinancialAssessment?.()]).then(() => {
+            if (isMounted()) {
+                setStateItems((previous_state: TWizardItemConfig[]) => {
+                    if (!previous_state.length) {
+                        return getItems(props);
+                    }
+                    return previous_state;
+                });
+                setPreviousData(fetchFromStorage());
+                setMounted(true);
+            }
+        });
+    }, [account_settings]);
+
+    React.useEffect(() => {
+        if (previous_data.length > 0) {
+            const items = [...state_items];
+            previous_data.forEach((item: any, index) => {
+                if (item instanceof Object) {
+                    items[index].form_value = item;
+                }
+            });
+            setStateItems(items);
+            setHasPreviousData(true);
+            setPreviousData([]);
+        }
+    }, [previous_data]);
+
+    const fetchFromStorage = () => {
+        const stored_items = localStorage.getItem('real_account_signup_wizard');
+        try {
+            const items = stored_items ? JSON.parse(stored_items) : {};
+            localStorage.removeItem('real_account_signup_wizard');
```

Comment:
Nit, but you could use a `finally` statement for this since it's duplicated in both `try` and `catch` statements.